datagateway_api/src/search_api/panosc_mappings.py

```diff
@@ -0,0 +1,52 @@
+import json
+import logging
+from pathlib import Path
+
+from datagateway_api.src.common.exceptions import SearchAPIError
+
+log = logging.getLogger()
+
+
+class PaNOSCMappings:
+    def __init__(
+        self, path=Path(__file__).parent.parent.parent / "search_api_mapping.json",
+    ):
+        """Load contents of `search_api_mapping.json` into this class"""
+        try:
+            with open(path, encoding="utf-8") as target:
+                log.info("Loading PaNOSC to ICAT mappings from %s", path)
+                self.mappings = json.load(target)
+        except IOError as e:
+            raise SearchAPIError(e)
```

Comment:
I think this a good idea, but a system exit should only occur if the user is intending to use the search API (they could just be using DataGateway API for example). I've added a conditional for that; I haven't added an `else` because this class should only really be used with the search API so probably wouldn't be reached in real-world conditions? Dealt with in d3687781fb880ce716618e9df94288ab2b45c25c

While testing the change, I also noticed and fixed a bug where there's an exception when there's no search API config in `config.json` but somewhere in the startup it looks for a piece of search API config.

Reply:
I agree